### PR TITLE
Feature/entrezgene parser

### PR DESCRIPTION
**Pull request by avullo**
## Requirements

- Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion;
- Review the [contributing guidelines](https://github.com/Ensembl/ensembl/blob/master/CONTRIBUTING.md#why-could-my-pull-request-be-rejected) for this repository; remember in particular:
    - do not modify code without testing for regression
    - provide simple unit tests to test the changes
    - if you change the schema you must patch the test databases as well, see [Updating the schema](https://github.com/Ensembl/ensembl/blob/master/CONTRIBUTING.md#updating-the-schema)
    - the PR must not fail unit testing

## Description

Port of the EntrezGene parser.

## Use case

Adding EntrezGene xrefs.

## Benefits

Having EntrezGene xrefs.

## Possible Drawbacks

None that I know.

## Testing

_Have you added/modified unit tests to test the changes?_

Yes, had to remove references to column 'ordered' in table source from various tests as this has been remove from the schema. Had to do that in order to successfully run the test suite.

_If so, do the tests pass/fail?_

Pass.

_Have you run the entire test suite and no regression was detected?_

Yes, except apache, perlcritic and pod coverage.

### CODE REVIEW

File: modules/Bio/EnsEMBL/Xref/Parser/EntrezGeneParser.pm [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+  # read data and load xrefs
+  my $xref_count = 0;
+  my $syn_count  = 0;
+  my %seen;    # record already processed xrefs
+
+  while ( my $data = $input_file->getline_hr( $eg_io ) ) {
+    # species_id corresponds to the species taxonomy id, see:
+    # https://github.com/Ensembl/ensembl-xref/pull/31#issuecomment-445838474
+    next unless $data->{'#tax_id'} eq "$species_id";
```
[the remark this answers is not in the document]
Comment (avullo) [reply modules/Bio/EnsEMBL/Xref/Parser/EntrezGeneParser.pm]: Species id is numeric but CSV fields aren't. Here I'm forcing string comparison with the proper operator as I wasn't sure about Perl doing the conversion.

File: modules/Bio/EnsEMBL/Xref/Parser/EntrezGeneParser.pm [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  my $wiki_source_id =
+    $xref_dba->get_source_id_for_source_name( "WikiGene", undef );
+
+  my $eg_io = $xref_dba->get_filehandle($file);
+  confess "Could not open $file" unless defined $eg_io;
+
+  my $input_file = Text::CSV->new(
+    { sep_char => "\t", empty_is_undef => 1, allow_loose_quotes => 1 } )
+    or
+    croak "Cannot use file $file: " . Text::CSV->error_diag();
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/EntrezGeneParser.pm]: confess !!

File: modules/Bio/EnsEMBL/Xref/Parser/EntrezGeneParser.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (nerdstrike) [reply modules/Bio/EnsEMBL/Xref/Parser/EntrezGeneParser.pm]: The use of a string comparator will force string context

File: modules/t/entrezgene.t [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+=cut
+
+use strict;
+use warnings;
+use Test::More;
+use Test::Exception;
+
+use FindBin '$Bin';
+use lib "$Bin/";
```
[the remark this answers is not in the document]
Comment (nerdstrike) [reply modules/t/entrezgene.t]: That's what FindBin does. Why do you need that folder in the PERL5LIB?

[the remark this answers is not in the document]
Comment (avullo) [reply modules/t/entrezgene.t]: You're right, oversight!

File: modules/Bio/EnsEMBL/Xref/Parser/EntrezGeneParser.pm (hunk 2 above)
Comment (avullo) [reply modules/Bio/EnsEMBL/Xref/Parser/EntrezGeneParser.pm]: True!

**Pull request merged**